server/src/main/java/org/elasticsearch/index/codec/bloomfilter/ES87BloomFilterPostingsFormat.java

```diff
@@ -0,0 +1,810 @@
+        abstract TermsEnum getDelegate() throws IOException;
+
+        @Override
+        public SeekStatus seekCeil(BytesRef text) throws IOException {
+            return getDelegate().seekCeil(text);
+        }
+
+        @Override
+        public void seekExact(long ord) throws IOException {
+            getDelegate().seekExact(ord);
+        }
+
+        @Override
+        public BytesRef term() throws IOException {
+            return getDelegate().term();
+        }
+
+        @Override
+        public long ord() throws IOException {
+            return getDelegate().ord();
+        }
+
+        @Override
+        public int docFreq() throws IOException {
+            return getDelegate().docFreq();
+        }
+
+        @Override
+        public long totalTermFreq() throws IOException {
+            return getDelegate().totalTermFreq();
+        }
+
+        @Override
+        public PostingsEnum postings(PostingsEnum reuse, int flags) throws IOException {
+            return getDelegate().postings(reuse, flags);
+        }
+
+        @Override
+        public ImpactsEnum impacts(int flags) throws IOException {
+            return getDelegate().impacts(flags);
+        }
+
+        @Override
+        public BytesRef next() throws IOException {
+            return getDelegate().next();
+        }
+
+        @Override
+        public AttributeSource attributes() {
+            try {
+                return getDelegate().attributes();
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+    }
+
+    static int bloomFilterSize(int maxDocs) {
+        if (maxDocs < 1) {
+            throw new IllegalStateException("maxDocs must be greater than or equal to 1, got " + maxDocs);
+        }
+        // 10% saturation (i.e., 10 bits for each term)
+        long numBits = maxDocs * (long) BITS_PER_ENTRY;
+        // Round to the next number of 8 since we can only store whole bytes
+        numBits = ((numBits - 1) | 0x07L) + 1;
+        if (numBits > Integer.MAX_VALUE) {
+            return Integer.MAX_VALUE;
+        } else {
+            return (int) numBits;
+        }
+    }
+
+    static int numBytesForBloomFilter(int bloomFilterSize) {
+        return Math.toIntExact((bloomFilterSize + 7L) / 8L);
+    }
+
+    // Uses MurmurHash3-128 to generate a 64-bit hash value, then picks 7 subsets of 31 bits each and returns the values in the
+    // outputs array. This provides us with 7 reasonably independent hashes of the data for the cost of one MurmurHash3 calculation.
+    // Note: We really only need a 64-bit value, but we are getting a 128-bit value back. We are discarding the 2nd value in this code.
+    // In theory, we could fold it into the hash functions, too, but it is not evident if it'd make a measurable difference.
+    // The extra allocation of a long[] is worrysome, we have to rely on either the JVM escape analysis or the nursery GCs to keep
```

Comment:
I made some changes moving the allocation out of the hash function itself, and into a small function that calls the hash function and later discards the object. I am unsure if this will be enough to entice the escape analysis to do the right thing, though, but I *think* it should.

Modifying the hash to only return a 64-bit value is a valid other choice. Your call on monday?

Reply:
It looks great to me this way.